Tests/AEPOptimizeTests/FunctionalTests/OptimizeFunctionalTests.swift

```diff
@@ -1021,13 +1021,15 @@ class OptimizeFunctionalTests: XCTestCase {
         let expectation = XCTestExpectation(description: "Get propositions request should now dispatch response event after update completion.")
         
         mockRuntime.onEventDispatch = { event in
-            expectation.fulfill()
+            if event.responseID == getEvent.id {
+                expectation.fulfill()
+            }
         }
         
-        wait(for: [expectation], timeout: 10)
-        XCTAssertEqual(mockRuntime.dispatchedEvents.count, 1)
+        wait(for: [expectation], timeout: 12)
+        XCTAssertEqual(mockRuntime.dispatchedEvents.count, 2)
         
-        let dispatchedEvent = mockRuntime.dispatchedEvents.first
```

Comment:
can you add assertions for the first event as well?

Reply:
Same comment for all following tests

Reply:
Done